src/audio/pipeline/pipeline-graph.c

```diff
@@ -95,6 +95,89 @@ static inline void pipeline_posn_offset_put(uint32_t posn_offset)
 	spin_unlock(&pipeline_posn->lock);
 }
 
+static void pipeline_devm_free_all(struct pipeline *pipe)
+{
+	struct pipe_dev_mem *mem;
+	struct list_item *mem_list;
+	struct list_item *_mem_list;
+
+	/* Free all memory in the list */
+	list_for_item_safe(mem_list, _mem_list, &pipe->mem.mem_list) {
+		mem = container_of(mem_list, struct pipe_dev_mem, mem_list);
+		rfree(mem->ptr);
+		list_item_del(&mem->mem_list);
+		rfree(mem);
+	}
+}
+
+void *pipeline_devm_alloc(struct pipeline *pipe, uint32_t size, uint32_t alignment)
```

Comment:
we need add caps for args since for ipc4 we need difference caps for different cases